File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        int n = score.length;
        int[] answer = new int[n];
        
        for (int i = 0; i < n; i++) {
            minHeap.offer(score[i]);
            if (minHeap.size() > k) {
                minHeap.poll();
            }
            answer[i] = minHeap.peek();
        }
        
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.kt
import java.util.*;

class Solution {
    fun solution(k: Int, score: IntArray): IntArray {
        val minHeap = PriorityQueue<Int>()
        val answer = IntArray(score.size)
        
        for (i in score.indices) {
            minHeap.offer(score[i])
            if (minHeap.size > k) {
                minHeap.poll()
            }
            answer[i] = minHeap.peek()
        }
        
        return answer
    }
}